### ISSUE TITLE
Crash Report: "ambiguous argument 'v1.7.8'"

### ISSUE THREAD

**Issue by SeanMyFiziq** [flutter/flutter#49604]
Reporting because the terminal output suggested I do so. Not expecting any outcome as I will uninstall Flutter and reinstall to hopefully resolve the issue. 

## Steps to Reproduce 

1. I ran the command: `flutter upgrade --force`
2. Crash

## Logs

```
Flutter crash report; please file at https://github.com/flutter/flutter/issues.

## command

flutter upgrade --force

## exception

ProcessException: ProcessException: Process "git" exited abnormally:
fatal: ambiguous argument 'v1.7.8': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
  Command: git reset --hard v1.7.8

#0      runCheckedAsync (package:flutter_tools/src/base/process.dart:259:7)
<asynchronous suspension>
#1      UpgradeCommandRunner.resetChanges (package:flutter_tools/src/commands/upgrade.dart:135:39)
<asynchronous suspension>
#2      UpgradeCommandRunner.runCommand (package:flutter_tools/src/commands/upgrade.dart:86:11)
<asynchronous suspension>
#3      UpgradeCommand.runCommand (package:flutter_tools/src/commands/upgrade.dart:47:32)
<asynchronous suspension>
#4      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:478:18)
<asynchronous suspension>
#5      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:383:33)
<asynchronous suspension>
#6      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:29)
<asynchronous suspension>
#7      _rootRun (dart:async/zone.dart:1124:13)
#8      _CustomZone.run (dart:async/zone.dart:1021:19)
#9      _runZoned (dart:async/zone.dart:1516:10)
#10     runZoned (dart:async/zone.dart:1463:12)
#11     AppContext.run (package:flutter_tools/src/base/context.dart:152:18)
<asynchronous suspension>
#12     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:375:20)
#13     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
<asynchronous suspension>
#14     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:396:21)
<asynchronous suspension>
#15     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:29)
<asynchronous suspension>
#16     _rootRun (dart:async/zone.dart:1124:13)
#17     _CustomZone.run (dart:async/zone.dart:1021:19)
#18     _runZoned (dart:async/zone.dart:1516:10)
#19     runZoned (dart:async/zone.dart:1463:12)
#20     AppContext.run (package:flutter_tools/src/base/context.dart:152:18)
<asynchronous suspension>
#21     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:356:19)
<asynchronous suspension>
#22     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#23     new Future.sync (dart:async/future.dart:224:31)
#24     CommandRunner.run (package:args/command_runner.dart:112:14)
#25     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:242:18)
#26     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:22)
<asynchronous suspension>
#27     _rootRun (dart:async/zone.dart:1124:13)
#28     _CustomZone.run (dart:async/zone.dart:1021:19)
#29     _runZoned (dart:async/zone.dart:1516:10)
#30     runZoned (dart:async/zone.dart:1500:12)
#31     run.<anonymous closure> (package:flutter_tools/runner.dart:60:18)
<asynchronous suspension>
#32     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:153:29)
<asynchronous suspension>
#33     _rootRun (dart:async/zone.dart:1124:13)
#34     _CustomZone.run (dart:async/zone.dart:1021:19)
#35     _runZoned (dart:async/zone.dart:1516:10)
#36     runZoned (dart:async/zone.dart:1463:12)
#37     AppContext.run (package:flutter_tools/src/base/context.dart:152:18)
<asynchronous suspension>
#38     runInContext (package:flutter_tools/src/context_runner.dart:56:24)
<asynchronous suspension>
#39     run (package:flutter_tools/runner.dart:51:10)
#40     main (package:flutter_tools/executable.dart:62:9)
<asynchronous suspension>
#41     main (file:///private/tmp/flutter-20190826-53895-i6kfgq/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#42     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#43     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

## flutter doctor

```
[33m[!][39m Flutter (Channel stable, v1.7.8+hotfix.4, on Mac OS X 10.15.2 19C57, locale
    en-AU)
    [32m•[39m Flutter version 1.7.8+hotfix.4 at /usr/local/Cellar/flutter/stable
    [32m•[39m Framework revision 20e59316b8 (6 months ago), 2019-07-18 20:04:33 -0700
    [32m•[39m Engine revision fee001c93f
    [32m•[39m Dart version 2.4.0
    [31m✗[39m Downloaded executables cannot execute on host.
      See https://github.com/flutter/flutter/issues/6207 for more information


[33m[!][39m Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    [32m•[39m Android SDK at /Users/myname/Library/Android/sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling
      support)
    [32m•[39m Platform android-29, build-tools 29.0.2
    [32m•[39m Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    [33m![39m Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses

[32m[✓][39m Xcode - develop for iOS and macOS (Xcode 11.3.1)
    [32m•[39m Xcode at /Applications/Xcode.app/Contents/Developer
    [32m•[39m Xcode 11.3.1, Build version 11C504
    [32m•[39m CocoaPods version 1.8.4

[33m[!][39m iOS tools - develop for iOS devices
    [31m✗[39m Verify that all connected devices have been paired with this computer in
      Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may
      require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    [32m•[39m ios-deploy 1.9.4

[32m[✓][39m Android Studio (version 3.5)
    [32m•[39m Android Studio at /Applications/Android Studio.app/Contents
    [32m•[39m Flutter plugin version 38.2.3
    [32m•[39m Dart plugin version 191.8423
    [32m•[39m Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[32m[✓][39m VS Code (version 1.41.1)
    [32m•[39m VS Code at /Applications/Visual Studio Code.app/Contents
    [32m•[39m Flutter extension version 3.7.1

[33m[!][39m Connected device
    [33m![39m No devices available

[33m![39m Doctor found issues in 4 categories.
```

`flutter doctor -v` output

```
[!] Flutter (Channel stable, v1.7.8+hotfix.4, on Mac OS X 10.15.2 19C57, locale
    en-AU)
    • Flutter version 1.7.8+hotfix.4 at /usr/local/Cellar/flutter/stable
    • Framework revision 20e59316b8 (6 months ago), 2019-07-18 20:04:33 -0700
    • Engine revision fee001c93f
    • Dart version 2.4.0
    ✗ Downloaded executables cannot execute on host.
      See https://github.com/flutter/flutter/issues/6207 for more information


[!] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/myname/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor
      --android-licenses

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.8.4

[!] iOS tools - develop for iOS devices
    ✗ Verify that all connected devices have been paired with this computer in
      Xcode.
      If all devices have been paired, libimobiledevice and ideviceinstaller may
      require updating.
      To update with Brew, run:
        brew update
        brew uninstall --ignore-dependencies libimobiledevice
        brew uninstall --ignore-dependencies usbmuxd
        brew install --HEAD usbmuxd
        brew unlink usbmuxd
        brew link usbmuxd
        brew install --HEAD libimobiledevice
        brew install ideviceinstaller
    • ios-deploy 1.9.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 38.2.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[!] Connected device
    ! No devices available

! Doctor found issues in 4 categories.
```

**Comment by VladyslavBondarenko** [comment]
Hi @SeanMyFiziq 
Does it help if you try to update with git?
https://github.com/flutter/flutter/wiki/Workarounds-for-common-issues#flutter-installation-corrupted

**Comment by SeanMyFiziq** [comment]
Hi, thanks for the prompt response. I have removed flutter and reinstalled following the steps found on [Flutter for MacOS](https://flutter.dev/docs/get-started/install/macos). I still get the same error though. If there were a simple way to remove all the flutter packages and dependencies so that a clean install could take place it would probably aid in resolving the issue.

**Comment by VladyslavBondarenko** [comment]
@SeanMyFiziq 
By 'removed flutter' do you mean you removed everything from flutter dir together with `.git` dir?

**Comment by SeanMyFiziq** [comment]
@VladyslavBondarenko - I believe I may have found the cause of this problem. I caused the problem - I had previously installed Flutter using `brew install flutter` early last year. 

I ran `brew uninstall flutter` reversed the instructions in following the [install Flutter on MacOS instructions](https://flutter.dev/docs/get-started/install/macos) (including removing `export PATH="$PATH:`pwd`/flutter/bin"`) and then reinstalled using brew. To complete the setup, I ran `flutter doctor`. 

All systems working without error now.
